### PR TITLE
Add dependecy watchers as codeowners of requirements

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

Could be a bad idea since this may result in lots of requested reviews for dependabot PRs, but maybe that also makes sense. 

The goal is to automate the process of raising awareness to the dependapod of dependency changes outside of dependabot PRs.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -61,16 +61,22 @@
...
 /docs/js-guide/ @orangejenny @millerdev
 
 /docker/ @millerdev @joxl
 
+/package.json @dimagi/dependency-watchers
+
 /scripts/docker @millerdev @joxl
 /scripts/vellum-to-hq @millerdev
 
+/requirements/ @dimagi/dependency-watchers
```
[the remark this answers is not in the document]
Comment (millerdev) [reply .github/CODEOWNERS:77]: If we do that we will not be tagged on dependabot PRs. Not sure if that's good or bad?

I might like it if I am not automatically tagged on every dependabot PR so I can easily discern the important ones when I am pulled into a discussion on one of those.

[the remark this answers is not in the document]
Comment (joxl) [reply .github/CODEOWNERS:77]: Ah yes, dependabot doesn't touch those. Definitely less noisy then. There are scenarios where human PRs could also avoid the `*.in` files (e.g. dependabot-like package upgrade), so for completeness perhaps leaving it wide open like it is now would be best?

[the remark this answers is not in the document]
Comment (gherceg) [reply .github/CODEOWNERS:77]: > I might like it if I am not automatically tagged on every dependabot PR so I can easily discern the important ones when I am pulled into a discussion on one of those.

I don't have an example at the moment to verify with, but is there a concrete difference that makes it easy to sort PRs where you as an individual are requested vs as part of the dependency watchers? Basically using the filter review-requested:gherceg vs review-requested:dimagi/dependency-watchers in GitHub. Granted if you rely on your email inbox you may need to setup additional filters there.

[the remark this answers is not in the document]
Comment (millerdev) [reply .github/CODEOWNERS:77]: I do use my email inbox, and might be able to construct a filter to get what I want. Currently PRs where my review has been requested are sent _To_ me, and that's how I know I definitely need to look at those PRs (I have a filter that adds a label to make them easy to spot).

I'm signed off on this as is, assuming the github permissions issue has been resolved.

**Review by millerdev — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
[the remark this answers is not in the document]
Comment (gherceg) [reply .github/CODEOWNERS:77]: Yeah the permissions issue has been resolved 👍🏻 and can confirm github is no longer flagging that error in this PR

**Pull request merged**